cli/flox/doc/manifest.toml.md

```diff
@@ -0,0 +1,310 @@
+---
+title: MANIFEST.TOML
+section: 1
+header: "Flox User Manuals"
+...
+
+
+# NAME
+
+manifest.toml - declarative environment configuration format
+
+# SYNOPSIS
+
+The `manifest.toml` file is a declarative format for specifying the packages
+installed in an environment,
+environment variables to make available to the environment,
+a shell script to run upon activation of the environment,
+and other options for how the environment should behave.
+
+# DESCRIPTION
+
+Flox environments come with a declarative manifest in TOML format.
+An environment can be reproduced from this one file.
+The file is divided into just a few sections that are represented as TOML
+tables:
+
+- `[install]`
+- `[vars]`
+- `[hook]`
+- `[options]`
+
+## `[install]`
+
+The `[install]` table is the core of the environment,
+specifying which packages you'd like installed in the environment.
+An example of the `[install]` table is shown below:
+
+```toml
+[install]
+ripgrep.pkg-path = "ripgrep"
+pip.pkg-path = "python310Packages.pip"
+```
+
+Since this is TOML, an equivalent way of writing this would be
+```toml
+[install]
+ripgrep = { pkg-path = "ripgrep" }
+pip = { pkg-path = "python310Packages.pip" }
+```
+Flox will use the first format by default when automatically editing
+the manifest.
+
+```{.include}
+./include/package-names.md
+```
+
+### Package descriptors
+
+Each entry in the `[install]` table is a key-value pair.
+The key in the key-value pair (e.g. `ripgrep`, `pip`) is referred to as an
+"install ID",
+and represents the name by which you will refer to a particular package e.g.
+if you wanted to uninstall or upgrade the package.
+Install IDs are inferred from the last attribute in the pkg-path,
+but may also be specified either at install-time via the `-i` option
+or interactively via `flox edit`.
+
+The value in the key-value pair is called a "package descriptor".
+A package is specified by a number of available options which are separate
+from the install ID,
+so you are free to change them independently of one another.
+This allows you to change package details while keeping a stable install ID,
+for example upgrading from `gcc.pkg-path = "gcc12"` to
+`gcc.pkg-path = "gcc13"`.
+
+The descriptor options allow you to specify in detail the package to install.
+The full list of descriptor options are shown below:
+```
+Descriptor ::= {
+  name               = null | <STRING>
+, optional           = null | <BOOL>
+, pkg-group          = null | <STRING>
+, version            = null | <STRING>
+, semver             = null | <STRING>
+, systems            = null | [<STRING>, ...]
+, pkg-path           = null | <STRING> | [<STRING>, ...]
+, abs-path           = null | <STRING> | [<STRING>, ...]
+, priority           = null | <INT>
+}
+```
+
+None of these options are required,
+and leaving them unset instructs Flox to simply find the best match for the
+package name and latest version given the install ID.
```

Comment:
nonblocking: do we use `iid` as `pkg-path`? I think saying that might be helpful. Why doesn't `flox install`  set `package = {}` if `package` doesn't contain `.`, just to be explicit?

Reply:
I documented the behavior and recommended against it.